src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let newObj = {
+    ...state,
+  };
+  const newArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newObj, action.extraData);
+        newArray.push({ ...newObj });
```

Comment:
You are pushing a clone of the 'newObj' into the 'newArray' repeatedly. This could be optimized by pushing the object only once after the switch statement.